### PR TITLE
Plenigo 666 noscript tag for wordpress

### PR DESCRIPTION
**Pull request by SebaSOFT**
Tested on Chrome and Firrefox

### CODE REVIEW

File: plenigo_plugin/PlenigoContentManager.php [hunk 1]
```diff
@@ -913,9 +919,38 @@ public function printGoogleAnalytics() {
     }
 
     /**
+     * Returns the &lt;noscript&gt; tag found in the plugin or theme directory
      * 
-     * @param string $strGAhtml The HTML to be replaced
+     * @return string the HTML to render or empty if no teplate has been found
```
Comment (robertholfeld) [inline plenigo_plugin/PlenigoContentManager.php]: Misspelling teplate

File: plenigo_plugin/PlenigoContentManager.php [hunk 2]
```diff
@@ -913,9 +919,38 @@ public function printGoogleAnalytics() {
...
+     * @param string $htmlText The template HTML text
+     * @return string The template HTML text with the replacement tags or empty if the NOSCRIPT functionality is disabled
      */
+    public function replace_noscript_tags($htmlText) {
+        $res = '';
+        if (isset($this->options['noscript_enabled']) && $this->options['noscript_enabled'] === 1) {
+            $strTitle = (isset($this->options['noscript_title'])) ? $this->options['noscript_title'] : __("You need JavaScript", self::PLENIGO_SETTINGS_GROUP);
+            $strMessage = (isset($this->options['noscript_message'])) ? $this->options['noscript_message'] : __("In order to provide you with the best experience, "
+                            . "this site requires that you allow JavaScript to run. "
+                            . "Please correct that and try again", self::PLENIGO_SETTINGS_GROUP);
```
Comment (robertholfeld) [inline plenigo_plugin/PlenigoContentManager.php]: Dot after again

File: plenigo_plugin/PlenigoSettingsPage.php [hunk 3]
```diff
@@ -148,35 +147,29 @@ public function add_help_tab()
...
         ));
         $screen->add_help_tab(array(
             'id' => 'plenigo_help_login',
             'title' => __('Plenigo OAuth Help', self::PLENIGO_SETTINGS_GROUP),
             'content' => '<p>' . __('In order to configure Plenigo OAuth Login: ', self::PLENIGO_SETTINGS_GROUP)
-            . __('1 - Add Login redirect URL to Plenigo (Usually: <b>{YOUR BLOG URL}/wp-login.php</b>) ',
-                self::PLENIGO_SETTINGS_GROUP)
-            . ' <a target="_blank" href="' . PlenigoContentManager::JS_BASE_URL_NOAUTH . '/company/account/urls/show">' . __('clicking this link',
-                self::PLENIGO_SETTINGS_GROUP) . '</a><br/>'
+            . __('1 - Add Login redirect URL to Plenigo (Usually: <b>{YOUR BLOG URL}/wp-login.php</b>) ', self::PLENIGO_SETTINGS_GROUP)
```
Comment (robertholfeld) [inline plenigo_plugin/PlenigoSettingsPage.php]: plenigo lower case

File: plenigo_plugin/PlenigoSettingsPage.php [hunk 4]
```diff
@@ -148,35 +147,29 @@ public function add_help_tab()
...
+            . ' <a target="_blank" href="' . PlenigoContentManager::JS_BASE_URL_NOAUTH . '/company/account/urls/show">' . __('clicking this link', self::PLENIGO_SETTINGS_GROUP) . '</a><br/>'
             . __('2 - Fill the same URL in the <b>OAuth redirect URL</b> below', self::PLENIGO_SETTINGS_GROUP) . '<br/>'
-            . __('3 - (Optional) Fill the URL in the <b>URL After Login</b> for login redirection',
-                self::PLENIGO_SETTINGS_GROUP) . '<br/>'
-            . __('4 - Enable the Plenigo Login clicking <b>Use Plenigo Authentication Provider</b> ',
-                self::PLENIGO_SETTINGS_GROUP) . '<br/>'
+            . __('3 - (Optional) Fill the URL in the <b>URL After Login</b> for login redirection', self::PLENIGO_SETTINGS_GROUP) . '<br/>'
+            . __('4 - Enable the Plenigo Login clicking <b>Use Plenigo Authentication Provider</b> ', self::PLENIGO_SETTINGS_GROUP) . '<br/>'
```
Comment (robertholfeld) [inline plenigo_plugin/PlenigoSettingsPage.php]: plenigo lower case

File: plenigo_plugin/PlenigoSettingsPage.php [hunk 5]
```diff
@@ -148,35 +147,29 @@ public function add_help_tab()
...
             . __('5 - Put the Plenigo Login Widget in a widget area of the site ', self::PLENIGO_SETTINGS_GROUP)
-            . ' <a target="_blank" href="' . admin_url('/widgets.php') . '">' . __('clicking this link',
-                self::PLENIGO_SETTINGS_GROUP) . '</a><br/>'
+            . ' <a target="_blank" href="' . admin_url('/widgets.php') . '">' . __('clicking this link', self::PLENIGO_SETTINGS_GROUP) . '</a><br/>'
```
Comment (robertholfeld) [inline plenigo_plugin/PlenigoSettingsPage.php:162]: plenigo lower case

File: plenigo_plugin/PlenigoSettingsPage.php [hunk 6]
```diff
@@ -309,56 +300,51 @@ public function sanitize($input)
...
+    public function print_section_general() {
         print '<div role="tabpanel" class="tab-pane active" id="plenigo_general">'
-            . '<h3>' . __('General', self::PLENIGO_SETTINGS_GROUP) . '</h3>'
-            . 'These are the basic settings for using plenigo Services. '
-            . 'It allow you to set your Company ID, your encryption secret code, '
-            . 'working in the test environment and also disabling '
-            . 'the entire plenigo functionality alltogether.';
+                . '<h3>' . __('General', self::PLENIGO_SETTINGS_GROUP) . '</h3>'
+                . 'These are the basic settings for using plenigo Services. '
+                . 'It allow you to set your Company ID, your encryption secret code, '
```
Comment (robertholfeld) [inline plenigo_plugin/PlenigoSettingsPage.php]: Dont know if it should be "It allow's you"

File: plenigo_plugin/PlenigoSettingsPage.php [hunk 7]
```diff
@@ -309,56 +300,51 @@ public function sanitize($input)
...
-    public function print_section_login()
-    {
+    public function print_section_login() {
         print '</div><div role="tabpanel" class="tab-pane active" id="plenigo_login_section">'
-            . '<h3>' . __('OAuth Login', self::PLENIGO_SETTINGS_GROUP) . '</h3>'
-            . "This section allows this Wordpress's users to login using plenigo authentication. "
-            . "The data will be stored in the Wordpress database. If you disable this, "
-            . "users may need to recover their passwords.";
+                . '<h3>' . __('OAuth Login', self::PLENIGO_SETTINGS_GROUP) . '</h3>'
+                . "This section allows this Wordpress's users to login using plenigo authentication. "
```
Comment (robertholfeld) [inline plenigo_plugin/PlenigoSettingsPage.php]: Allows the Wordpress's users

File: plenigo_plugin/settings/SettingNoscriptMessage.php [hunk 8]
```diff
@@ -0,0 +1,67 @@
+<?php
+
+/*
+  Copyright (C) 2014 Plenigo
```
Comment (robertholfeld) [inline plenigo_plugin/settings/SettingNoscriptMessage.php]: Plenigo lower case

File: plenigo_plugin/settings/SettingNoscriptMessage.php [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+    /**
+     * @see PlenigoWPSetting::getDefaultValue()
+     */
+    public function getDefaultValue($current = null) {
+        if (!is_null($current)) {
+            return $current;
+        }
+        return __("In order to provide you with the best experience, "
+                . "this site requires that you allow JavaScript to run. "
+                . "Please correct that and try again", self::PLENIGO_SETTINGS_GROUP);
```
Comment (robertholfeld) [inline plenigo_plugin/settings/SettingNoscriptMessage.php]: Dot after again

File: plenigo_plugin/settings/SettingNoscriptMessage.php [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+    }
+
+    /**
+     * @see PlenigoWPSetting::getTitle()
+     */
+    public function getTitle() {
+        return __('no-Javacript overlay Mesage', parent::PLENIGO_SETTINGS_GROUP);
```
Comment (robertholfeld) [inline plenigo_plugin/settings/SettingNoscriptMessage.php]: message

File: plenigo_plugin/settings/SettingNoscriptTitle.php [hunk 11]
```diff
@@ -0,0 +1,56 @@
+<?php
+
+/*
+  Copyright (C) 2014 Plenigo
```
Comment (robertholfeld) [inline plenigo_plugin/settings/SettingNoscriptTitle.php]: Plenigo lowercase

File: plenigo_plugin/settings/SettingUseNoscript.php [hunk 12]
```diff
@@ -0,0 +1,95 @@
+<?php
+
+/*
+  Copyright (C) 2014 Plenigo
```
Comment (robertholfeld) [inline plenigo_plugin/settings/SettingUseNoscript.php]: Plenigo lowercase

File: plenigo_plugin/settings/SettingUseNoscript.php [hunk 13]
```diff
@@ -0,0 +1,95 @@
...
+     * Returns the title of the ON option
+     * 
+     * @return string
+     */
+    protected function getOnTitle() {
+        return __('Show the no-Javacript overlay', parent::PLENIGO_SETTINGS_GROUP);
+    }
+
+    /**
+     * Returns the tiele of the OFF option
```
Comment (robertholfeld) [inline plenigo_plugin/settings/SettingUseNoscript.php]: Misspelling tiele

File: readme.txt [hunk 14]
```diff
@@ -94,6 +94,9 @@ A: Because we make Chuck Norris eat all his meal
 Coming soon
 
 == Changelog ==
+= 1.1.25 - no-Javascript overlay =
+- Feature: Plenigo works best with Javascript enable. Now we provide you a nice overlay you can customize (or event theme it yourself) to show your users that the best experience is enabling Javascript.
```
Comment (robertholfeld) [inline readme.txt]: Javascript enabled.

File: plenigo_plugin/PlenigoContentManager.php (hunk 1 above)
Comment (SebaSOFT) [inline plenigo_plugin/PlenigoContentManager.php]: done

File: plenigo_plugin/PlenigoContentManager.php (hunk 2 above)
Comment (SebaSOFT) [inline plenigo_plugin/PlenigoContentManager.php]: done

File: plenigo_plugin/PlenigoSettingsPage.php (hunk 3 above)
Comment (SebaSOFT) [inline plenigo_plugin/PlenigoSettingsPage.php]: done

File: plenigo_plugin/PlenigoSettingsPage.php (hunk 4 above)
Comment (SebaSOFT) [inline plenigo_plugin/PlenigoSettingsPage.php]: done

File: plenigo_plugin/PlenigoSettingsPage.php (hunk 5 above)
Comment (SebaSOFT) [inline plenigo_plugin/PlenigoSettingsPage.php:162]: done

File: plenigo_plugin/PlenigoSettingsPage.php (hunk 6 above)
Comment (SebaSOFT) [inline plenigo_plugin/PlenigoSettingsPage.php]: done

File: plenigo_plugin/PlenigoSettingsPage.php (hunk 7 above)
Comment (SebaSOFT) [inline plenigo_plugin/PlenigoSettingsPage.php]: done

File: plenigo_plugin/settings/SettingNoscriptMessage.php (hunk 8 above)
Comment (SebaSOFT) [inline plenigo_plugin/settings/SettingNoscriptMessage.php]: done on all copyrights

File: plenigo_plugin/settings/SettingNoscriptMessage.php (hunk 9 above)
Comment (SebaSOFT) [inline plenigo_plugin/settings/SettingNoscriptMessage.php]: done

File: plenigo_plugin/settings/SettingNoscriptMessage.php (hunk 10 above)
Comment (SebaSOFT) [inline plenigo_plugin/settings/SettingNoscriptMessage.php]: Good catch

File: plenigo_plugin/settings/SettingNoscriptTitle.php (hunk 11 above)
Comment (SebaSOFT) [inline plenigo_plugin/settings/SettingNoscriptTitle.php]: done on all copyrights

File: plenigo_plugin/settings/SettingUseNoscript.php (hunk 12 above)
Comment (SebaSOFT) [inline plenigo_plugin/settings/SettingUseNoscript.php]: done on all copyrights

File: plenigo_plugin/settings/SettingUseNoscript.php (hunk 13 above)
Comment (SebaSOFT) [inline plenigo_plugin/settings/SettingUseNoscript.php]: done

File: readme.txt (hunk 14 above)
Comment (SebaSOFT) [inline readme.txt]: done

**Pull request merged**